### DISCUSSION TITLE
How GitHub maintains its advisory database

### DISCUSSION THREAD

**Discussion by nasifimtiazohi** [community/community#23167] [New to GitHub]
<p>Hi,</p>
<p>I am a PhD Student at NC State University. As part of our research, we are evaluating the existing tools that detect vulnerable dependencies. We have observed that the tools’ result can differ based on the strength of its vulnerability database. For our research, we are hoping to understand more on how services like GitHub maintains its vulnerability database. We’d be grateful if we get some responses for the below questions.</p>
<ol>
<li>What are your sources for vulnerability data, e.g. NVD, OSS Index?</li>
<li>Do you have any process to discover open source vulnerabilities by yourselves, e.g. through monitoring bug repositories? If yes, is it possible to share with us a high level explanation of what you do?</li>
<li>When collecting vulnerability data from third-party databases (e.g. NVD), do you perform any curation and/or correction, e.g. discarding debated CVEs or correcting the affected version range? If yes, is it possible to share with us a high level explanation of what you do?</li>
</ol>
<p>Thanks,<br>
Nasif</p>

**Comment by mayakacz** [comment, marked as answer]
<blockquote>
<ol>
<li>What are your sources for vulnerability data, e.g. NVD, OSS Index?</li>
</ol>
</blockquote>
<p>See the sources listed in our documentation: <a href="https://docs.github.com/en/github/managing-security-vulnerabilities/browsing-security-vulnerabilities-in-the-github-advisory-database#about-the-github-advisory-database" class="inline-onebox" rel="noopener nofollow ugc">Browsing security vulnerabilities in the GitHub Advisory Database - GitHub Docs</a></p>
<blockquote>
<ol start="2">
<li>Do you have any process to discover open source vulnerabilities by yourselves, e.g. through monitoring bug repositories? If yes, is it possible to share with us a high level explanation of what you do?</li>
</ol>
</blockquote>
<p>Yes, though most of our time is spent on curation and helping maintainers report their advisories.</p>
<blockquote>
<ol start="3">
<li>When collecting vulnerability data from third-party databases (e.g. NVD), do you perform any curation and/or correction, e.g. discarding debated CVEs or correcting the affected version range? If yes, is it possible to share with us a high level explanation of what you do?</li>
</ol>
</blockquote>
<p>Yes. See a description in this blog post: <a href="https://github.blog/2019-12-11-behind-the-scenes-github-vulnerability-alerts/" class="inline-onebox" rel="noopener nofollow ugc">Behind the scenes: GitHub security alerts - The GitHub Blog</a></p>
